cli/src/base/ContentDirectoryCommandBase.ts

```diff
@@ -109,6 +121,75 @@ export default abstract class ContentDirectoryCommandBase extends WorkingGroupCo
       : actor.isMember && actor.asMember.eq(channel.owner.asMember)
   }
 
+  async hasRequiredChannelAgentPermissions(
+    actor: ContentActor,
+    channel: Channel,
+    requiredPermissions: ChannelActionPermission['type'][]
+  ): Promise<boolean> {
+    // CASE: CuratorGroup owned channel
+    if (channel.owner.isCuratorGroup) {
+      // Lead context
+      if (actor.isLead) {
+        return true
+      }
+      // Curator context
+      if (actor.isCurator && actor.asCurator[0].eq(channel.owner.asCuratorGroup)) {
+        const { curators } = await this.getCuratorGroup(actor.asCurator[0].toNumber())
+        const curatorChannelAgentPermissions = curators.get(actor.asCurator[1])
+        return !!(
+          curatorChannelAgentPermissions &&
+          requiredPermissions.every((requiredPermission) =>
+            [...curatorChannelAgentPermissions].find((p) => p[`is${requiredPermission}`])
+          )
+        )
+      }
+      // Collaborator context
+      const collaboratorChannelAgentPermissions = channel.collaborators.get(actor.asMember)
+      return !!(
+        collaboratorChannelAgentPermissions &&
+        requiredPermissions.every((requiredPermission) =>
+          [...collaboratorChannelAgentPermissions].find((p) => p[`is${requiredPermission}`])
+        )
+      )
+    }
+
+    // CASE: Member owned channel
+    if (channel.owner.isMember) {
+      // Owner context
+      if (actor.isMember) {
+        return actor.asMember.eq(channel.owner.asMember)
+      }
+      // Collaborator context
+      const collaboratorChannelAgentPermissions = channel.collaborators.get(actor.asMember)
+      return !!(
+        collaboratorChannelAgentPermissions &&
+        requiredPermissions.every((requiredPermission) =>
+          [...collaboratorChannelAgentPermissions].find((p) => p[`is${requiredPermission}`])
+        )
+      )
+    }
+
+    return false
+  }
+
+  async isModeratorWithRequiredPermission(
+    actor: ContentActor,
+    channelPrivilegeLevel: ChannelPrivilegeLevel,
+    permission: CreateInterface<ContentModerationAction>
+  ): Promise<boolean> {
+    if (actor.isLead) {
+      return true
+    }
+
+    const permissionRequired = createType('PalletContentPermissionsCuratorGroupContentModerationAction', permission)
+    const { permissionsByLevel } = await this.getCuratorGroup(actor.asCurator[0].toNumber())
+    const permissionsForLevel = permissionsByLevel.get(channelPrivilegeLevel)
```

Comment:
Same issue with `BTreeMap.get`:
```suggestion
    const permissionsForLevel = [...permissionsByLevel].find(([k]) => k.eq(channelPrivilegeLevel))?.[1]
```